lib/ansible/inventory/manager.py

```diff
@@ -69,9 +69,9 @@ def order_patterns(patterns):
     pattern_intersection = []
     pattern_exclude = []
     for p in patterns:
-        if p[0] == "!":
+        if p and p[0] == "!":
```

Comment:
@sivel  So this was introduced in https://github.com/ansible/ansible/pull/58400 and there are more changes where we replaced `startswith` with checking the first index instead for perf purposes. I wonder if doing `p and p[0]` over `startswith` is still beneficial. Or, if we should filter empty patterns someplace up, in the code that calls this.

Reply:
@mkrizek see my comment above, checking `if not p` would still be a net gain

Reply:
@bcoca +1, I should've refreshed before commenting ;-)

Still, not for this PR but we should audit https://github.com/ansible/ansible/pull/58400 for similar possibilities of `IndexError`.